analysis/behavioral_analysis/cannabis_table.ipynb

```diff
@@ -0,0 +1,228 @@
+    "    \n",
+    "    \n",
+    "    \n",
+    "    for orig_name, col_name in by_ses_additions:\n",
+    "             \n",
+    "        dict_MM_baseline = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'Baseline'].groupby('IDS.CHR.Subject')[orig_name].agg(\"first\").to_dict()\n",
+    "        dict_MM_baseline = {f'{sub}_baseline':val for (sub,val) in dict_MM_baseline.items() if 'MM' in sub}\n",
+    "        \n",
+    "        dict_MM_1year = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'One year'].groupby('IDS.CHR.Subject')[orig_name].agg(\"first\").to_dict()\n",
+    "        dict_MM_1year = {f'{sub}_1year':val for (sub,val) in dict_MM_1year.items() if 'MM' in sub}\n",
+    "\n",
+    "        \n",
+    "        dict_MM = {**dict_MM_baseline, **dict_MM_1year}\n",
+    "        df_subs[col_name] = df_subs['sub_ses'].map(dict_MM)#.fillna(df_subs[col_name])\n",
+    "\n",
+    "    \n",
+    "    return df_subs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0dc25789",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def save_table1(df_subs,include_pval):\n",
+    "    \n",
+    "\n",
+    "    columns = ['CUD diagnosis','CUDIT summed score','THC frequency per month','Positive urine THC']\n",
+    "    categorical = ['CUD diagnosis','THC frequency per month','Positive urine THC']\n",
+    "    \n",
+    "    groupby = ['ses']\n",
+    "    labels={'MM_baseline': 'MC baseline','MM_1year': 'MC one-year'}\n",
+    "    \n",
+    "    if include_pval:\n",
+    "        mytable = TableOne(df_subs, columns=columns, categorical=categorical, groupby=groupby, rename=labels, pval=True)\n",
+    "    else:\n",
+    "        mytable = TableOne(df_subs, columns=columns, categorical=categorical, groupby=groupby, rename=labels, pval=False)\n",
+    "\n",
+    "    \n",
+    "    #create paths to output dir if not exist\n",
+    "    derivatives_path = '../../../derivatives'\n",
+    "    nilearn_output_path = os.path.join(derivatives_path, 'demographics')\n",
+    "    if not os.path.isdir(nilearn_output_path):\n",
+    "        os.makedirs (nilearn_output_path)\n",
+    "    \n",
+    "    if include_pval:\n",
+    "        mytable.to_csv('../../../derivatives/demographics/cannabis_table1_all_pval.csv')                          \n",
+    "        mytable_df = pd.read_csv('../../../derivatives/demographics/cannabis_table1_all_pval.csv')\n",
+    "        mytable_df.drop('Grouped by ses', axis=1, inplace=True)    #remove missing\n",
+    "        mytable_df.drop('Grouped by ses.1', axis=1, inplace=True)  #remove overall\n",
+    "        mytable_df = mytable_df[['Unnamed: 0','Unnamed: 1','Grouped by ses.3','Grouped by ses.2','Grouped by ses.4']] #swap 1 year and baseline columns\n",
+    "        mytable_df.to_csv('../../../derivatives/demographics/cannabis_table1_all_pval.csv',index=False)\n",
```

Comment:
because the format of mytable is not a dataframe that i can edit (i think it's a class in the table 1 package), so i the only way i was able to extract a pandas df from it was to save it as csv and then read it as a pandas df. then i edit and save again. i agree, it's not the nicest, but it was the most practical for what i wanted to do